modules/deployment/src/main/java/org/mule/runtime/module/deployment/internal/DefaultArchiveDeployer.java

```diff
@@ -309,13 +316,15 @@ private void undeploy(T artifact, boolean removeData) {
       deployer.undeploy(artifact);
       artifactArchiveInstaller.uninstallArtifact(artifact.getArtifactName());
       if (removeData) {
-        final File dataFolder = getAppDataFolder(artifact.getDescriptor().getDataFolderName());
+        final String appName = artifact.getDescriptor().getDataFolderName();
```

Comment:
This is not the appName but the change is not even necessary so you can restore the original line.

final File dataFolder = getAppDataFolder(artifact.getDescriptor().getDataFolderName());